pyuvsim/simsetup.py

```diff
@@ -100,80 +138,83 @@ def array_to_sourcelist(catalog_table, lst_array=None, time_array=None, latitude
+
     sourcelist = []
-    if len(catalog_table.shape) == 0:
-        catalog_table = catalog_table.reshape((1))
     Nsrcs = catalog_table.shape[0]
-    coarse_kwds = [lst_array, latitude_deg]
-    coarse_horizon_cut = all([k is not None for k in coarse_kwds])
-    if (not coarse_horizon_cut) and any([k is not None for k in coarse_kwds]):
-        warnings.warn("It looks like you want to do a coarse horizon cut, but you're missing keywords!")
+    coarse_horizon_cut = latitude_deg is not None
 
     if coarse_horizon_cut:
         lat_rad = np.radians(latitude_deg)
         buff = horizon_buffer
-    for (source_id, ra_j2000, dec_j2000, flux_I, freq) in catalog_table:
-        if min_flux:
-            if (flux_I < min_flux):
-                continue
-        if max_flux:
-            if (flux_I > max_flux):
-                continue
-        ra = Angle(ra_j2000, units.deg)
-        dec = Angle(dec_j2000, units.deg)
-        rise_lst = None
-        set_lst = None
-        if coarse_horizon_cut:
-            # Identify circumpolar sources and unrising sources.
-            tans = np.tan(lat_rad) * np.tan(dec.rad)
-            if tans < -1:
-                continue   # Source doesn't rise.
-            circumpolar = tans > 1
-
-            if not circumpolar:
-                rise_lst = ra.rad - np.arccos((-1) * tans)
-                set_lst = ra.rad + np.arccos((-1) * tans)
-                rise_lst -= buff
-                set_lst += buff
-
-                if rise_lst < 0:
-                    rise_lst += 2 * np.pi
-                if set_lst > 2 * np.pi:
-                    set_lst -= 2 * np.pi
-
-        source = Source(source_id, ra,
-                        dec,
-                        freq=freq * units.Hz,
-                        stokes=np.array([flux_I, 0., 0., 0.]),
-                        rise_lst=rise_lst,
-                        set_lst=set_lst)
-
-        sourcelist.append(source)
-
-    return np.array(sourcelist)
-
-
-def read_votable_catalog(gleam_votable, input_uv=None, source_select_kwds={}):
+
+    if min_flux:
+        catalog_table = catalog_table[catalog_table['flux_density_I'] > min_flux]
+
+    if max_flux:
+        catalog_table = catalog_table[catalog_table['flux_density_I'] < max_flux]
+
+    ra = Angle(catalog_table['ra_j2000'], units.deg)
+    dec = Angle(catalog_table['dec_j2000'], units.deg)
+
+    if coarse_horizon_cut:
+        tans = np.tan(lat_rad) * np.tan(dec.rad)
+        nonrising = tans < -1
+
+        catalog_table = catalog_table[~nonrising]
+        tans = tans[~nonrising]
+
+        ra = ra[~nonrising]
+        dec = dec[~nonrising]
+
+    ids = catalog_table['source_id']
+    freqs = catalog_table['frequency']
+
+    if coarse_horizon_cut:
+        circumpolar = tans >= 1      # These will have rise/set lst set to nan
+        with warnings.catch_warnings():
+            warnings.simplefilter('ignore', RuntimeWarning)
```

Comment:
This will catch all runtime warnings, is that what you want? I presume there's a specific warning you're trying to catch, might be good to make this more specific. warnings.filterwarnings can take a message.